pkg/api/validation/validation.go

```diff
@@ -4389,6 +4395,12 @@ func ValidateSecurityContext(sc *api.SecurityContext, fldPath *field.Path) field
 		}
 	}
 
+	if sc.RunAsGroup != nil {
+		if *sc.RunAsGroup < 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("runAsGroup"), *sc.RunAsGroup, isNegativeErrorMsg))
```

Comment:
That's looks odd that we're validating GID with `validation.IsValidGroupID` in one case and by using simple comparison on another. Let's use `validation.IsValidGroupID` in both cases as it covers more cases.